xrootd/xrootd_config.go

```diff
@@ -316,7 +316,17 @@ func CheckCacheXrootdEnv(exportPath string, server server_utils.XRootDServer, ui
 	if err != nil {
 		return "", errors.Wrap(err, "Failed to pull information from the federation")
 	}
-	viper.Set("Cache.PSSOrigin", param.Federation_DirectorUrl.GetString())
+	if directorUrlStr := param.Federation_DirectorUrl.GetString(); directorUrlStr != "" {
+		directorUrl, err := url.Parse(param.Federation_DirectorUrl.GetString())
+		if err == nil {
+			log.Debugln("Parsing director URL for 'pss.origin' setting:", directorUrlStr)
+			directorUrl.Scheme = "pelican"
+			viper.Set("Cache.PSSOrigin", directorUrl.String())
+		} else {
+			return "", errors.Wrapf(err, "Failed to parse director URL %s", directorUrlStr)
+		}
+	}
+
```

Comment:
In the `Federation.DiscoveryUrl` parsing below, similarly generate an error if the path component is not trivial.

Reply:
Also, I think we should switch the logic order here.  If DirectorUrl is set, it is more specific so it should override the DiscoveryUrl.

Reply:
We don't want an error in this case. The discoveryUrl.Host is gotten from discoveryUrl.Path. So the host shouldn't be trivial.

Reply:
Ah, wait, I see the way to check